app/src/main/java/com/jerboa/ui/components/community/list/CommunityListViewModel.kt

```diff
@@ -4,34 +4,99 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
+import com.jerboa.DEBOUNCE_DELAY
 import com.jerboa.api.API
 import com.jerboa.api.ApiState
 import com.jerboa.api.apiWrapper
+import com.jerboa.datatypes.types.Community
 import com.jerboa.datatypes.types.CommunityAggregates
 import com.jerboa.datatypes.types.CommunityView
 import com.jerboa.datatypes.types.Search
 import com.jerboa.datatypes.types.SearchResponse
 import com.jerboa.datatypes.types.SearchType
+import com.jerboa.datatypes.types.SortType
 import com.jerboa.datatypes.types.SubscribedType
+import com.jerboa.db.SearchHistory
+import com.jerboa.db.SearchHistoryRepository
 import com.jerboa.serializeToMap
 import com.jerboa.ui.components.common.Initializable
 import com.jerboa.ui.components.home.SiteViewModel
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
+import java.time.Instant
 
-class CommunityListViewModel : ViewModel(), Initializable {
-    override var initialized by mutableStateOf(false)
+class CommunityListViewModelFactory(private val repository: SearchHistoryRepository) :
```

Comment:
No, just make the searchhistory its own viewmodel like the other examples, and pass in either the viewmodel, or the `searchHistoryViewModel.observeAsState()` into the activities.